In [4]:
## Imports ##
import pandas as pd
import json

### Template

In [5]:
template = {
    "uri" : "",
    "raw" : "",
    "convertedLongitude" : 0,
    "convertedLatitude" : 0
}

# Stores info for all loci, exported as JSON
all_loci = []

### Do The Thing (TM)

In [6]:
print("Type \"x\" to end.")

while True:
    try:
        new_locus = template.copy()

        # URI entry
        print("-NEW LOCUS-")
        uri = input("Please enter a locus URI: ")
        if uri == "x":
            raise Exception()
        new_locus["uri"] = uri

        # Coords entry
        while True:
            raw = input("Please enter the raw coords of a new point in the format \"{PointLabel}: {NScoord}/{EWcoord}\": ")
            if raw == "x":
                break
            new_locus["raw"] = raw

            new_locus["convertedLongitude"] = float(input(f"Please enter the converted longitude of this point: "))
            new_locus["convertedLatitude"] = float(input(f"Please enter the converted latitude of this point: "))

            all_loci.append(new_locus.copy())
        
    except:
        print("Process finished.")
        print(all_loci)
        break

            

Type "x" to end.
-NEW LOCUS-
-NEW LOCUS-
-NEW LOCUS-
-NEW LOCUS-
-NEW LOCUS-
Process finished.
[{'uri': 'https://opencontext.org/subjects/454d0497-3ac5-446a-87b6-8ea516d5c7ca', 'raw': 'NW: 209.8, 19.32', 'convertedLongitude': 11.4021983, 'convertedLatitude': 43.1533638}, {'uri': 'https://opencontext.org/subjects/454d0497-3ac5-446a-87b6-8ea516d5c7ca', 'raw': 'A: 209.6, 18.6', 'convertedLongitude': 11.4021952, 'convertedLatitude': 43.1533575}, {'uri': 'https://opencontext.org/subjects/454d0497-3ac5-446a-87b6-8ea516d5c7ca', 'raw': 'B: 209.6, 17.75', 'convertedLongitude': 11.4021945, 'convertedLatitude': 43.1533499}, {'uri': 'https://opencontext.org/subjects/454d0497-3ac5-446a-87b6-8ea516d5c7ca', 'raw': 'SE: 210, 17', 'convertedLongitude': 11.4021987, 'convertedLatitude': 43.1533429}, {'uri': 'https://opencontext.org/subjects/454d0497-3ac5-446a-87b6-8ea516d5c7ca', 'raw': 'SW: 208.76, 17', 'convertedLongitude': 11.4021835, 'convertedLatitude': 43.1533437}, {'uri': 'https://opencontext.org/s

Stuff to use


https://opencontext.org/subjects/454d0497-3ac5-446a-87b6-8ea516d5c7ca

NW: 209.8, 19.32

A: 209.6, 18.6

B: 209.6, 17.75

SE: 210, 17

SW: 208.76, 17




https://opencontext.org/subjects/f4bcc749-10c4-4dd6-92a5-318eb21ef8ff

NW: 171, -40.28

NE: 171.40, -40.60

SE: 171.12, -41.80

SW: 170.56, -41.40




https://opencontext.org/subjects/99aa13ef-3e75-4dcf-ab40-db439452145d

NW: 173, 59.70

NE: 176, 59.70

SE: 176, 60.70

SW: 173, 60.70




https://opencontext.org/subjects/7815e41d-e70f-4580-a6be-7d2b435592c4

NW: 174, 59.90

NE: 176, 59.90

SE: 176, 64

SW: 174, 64




https://opencontext.org/subjects/ea97bafb-5633-4e93-ac0f-017a2870994c

NW: 208, 16.16

SE: 210.65, 15

SW: 208, 15




https://opencontext.org/subjects/b2547403-6444-4310-9b58-9786f5c9e818

NW: 208.77, 17

NE1: 212, 17

NE2: 211.82, 16.34

SE1: 211.36, 15.89

SE2: 211.16, 15.58





In [7]:
## Saving results ##
filename = "LociWithCoordsConverted"
with open(filename, 'w') as f:
    f.write(json.dumps(all_loci))
print(len(all_loci))

21
